postgresql/src/main/java/org/apache/zeppelin/postgresql/PostgreSqlInterpreter.java

```diff
@@ -123,10 +145,16 @@ public void open() {
 
       jdbcConnection = DriverManager.getConnection(url, user, password);
 
+      Set<String> keywordsCompletions = SqlCompleter.getSqlKeywordsCompletions(jdbcConnection);
+      Set<String> dataModelCompletions =
+          SqlCompleter.getDataModelMetadataCompletions(jdbcConnection);
+      SetView<String> allCompletions = Sets.union(keywordsCompletions, dataModelCompletions);
+      sqlCompleter = new SqlCompleter(allCompletions, dataModelCompletions);
+
       exceptionOnConnect = null;
       logger.info("Successfully created psql connection");
 
-    } catch (ClassNotFoundException | SQLException e) {
+    } catch (ClassNotFoundException | SQLException | IOException e) {
```

Comment:
How about separating catch for SQLException and the rest?

In the other catch exceptions we check for null for the jdbcConnection and close it?